bclientapi/bendoapi.go

```diff
@@ -23,7 +23,11 @@ var (
 func (ia *itemAttributes) GetItemInfo() (*jason.Object, error) {
 	var path = "http://" + ia.bendoServer + "/item/" + ia.item
 
-	r, err := http.Get(path)
+	req, _ := http.NewRequest("GET", path, nil)
+        if ia.token != "" { req.Header.Add("X-Api-Key", ia.token) }
```

Comment:
Geeze- that looks like something a Ruby programmer would do 🙄

On Thu, Feb 25, 2016 at 3:54 PM, Don Brower <notifications@github.com>
wrote:

> In bclientapi/bendoapi.go
> <https://github.com/ndlib/bendo/pull/92#discussion_r54162631>:
>
> > @@ -23,7 +23,11 @@ var (
> >  func (ia *itemAttributes) GetItemInfo() (*jason.Object, error) {
> >  	var path = "http://" + ia.bendoServer + "/item/" + ia.item
> >
> > -	r, err := http.Get(path)
> > +	req, _ := http.NewRequest("GET", path, nil)
> > +        if ia.token != "" { req.Header.Add("X-Api-Key", ia.token) }
>
> There is a lot of commonality making requests to the bendo server between
> GetItemInfo(), download(), PostUpload(), createFileTransAction(), and
> getTransactionStatus(). Any more and it will be time to make a utility
> function (cf.
> https://github.com/ndlib/bendo/blob/master/store/cow.go#L215-L224).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ndlib/bendo/pull/92/files#r54162631>.
>